File: Kirameku-backend/admin/src/api/acg.ts
import { http } from "@/utils/http";

export type AcgStatus = "watched" | "watching" | "plan" | "on_hold" | "dropped";

export type AcgPersonalFields = {
  status: AcgStatus;
  progress: number;
  personal_score: number;
  review: string;
  watched_at: string;
  favorite: boolean;
  is_public: boolean;
  sort: number;
};

export type AcgItem = AcgPersonalFields & {
  id: number;
  bangumi_id: number;
  name: string;
  name_cn: string;
  cover_url: string;
  summary: string;
  air_date: string;
  year: number;
  total_episodes: number;
  bangumi_score: number;
  bangumi_rank: number;
  tags: string[];
  source_url: string;
  created_at: string;
  updated_at: string | null;
};

export type BangumiSearchItem = {
  bangumi_id: number;
  name: string;
  name_cn: string;
  cover_url: string;
  summary: string;
  air_date: string;
  year: number;
  total_episodes: number;
  score: number;
  rank: number;
  tags: string[];
  source_url: string;
};

export const getAcgItems = () =>
  http.request<AcgItem[]>("get", "/api/acg/admin");

export const searchBangumi = (query: string) =>
  http.request<BangumiSearchItem[]>("get", "/api/acg/search", {
    params: { q: query }
  });

export const importBangumiItem = (bangumiId: number, data: AcgPersonalFields) =>
  http.request<AcgItem>("post", `/api/acg/import/${bangumiId}`, { data });

export const updateAcgItem = (
  itemId: number,
  data: Partial<AcgPersonalFields>
) => http.request<AcgItem>("put", `/api/acg/${itemId}`, { data });

export const refreshAcgItem = (itemId: number) =>
  http.request<AcgItem>("post", `/api/acg/${itemId}/refresh`);

export const deleteAcgItem = (itemId: number) =>
  http.request<{ ok: boolean }>("delete", `/api/acg/${itemId}`);

File: Kirameku-backend/app/schemas/acg.py
from datetime import datetime

from pydantic import BaseModel, Field


class AcgPersonalFields(BaseModel):
    status: str = "watched"
    progress: int = Field(default=0, ge=0)
    personal_score: float = Field(default=0, ge=0, le=10)
    review: str = ""
    watched_at: str = ""
    favorite: bool = False
    is_public: bool = True
    sort: int = 0


class AcgItemCreate(AcgPersonalFields):
    bangumi_id: int
    name: str
    name_cn: str = ""
    cover_url: str = ""
    summary: str = ""
    air_date: str = ""
    year: int = 0
    total_episodes: int = 0
    bangumi_score: float = 0
    bangumi_rank: int = 0
    tags: list[str] = Field(default_factory=list)
    source_url: str = ""


class AcgItemUpdate(BaseModel):
    status: str | None = None
    progress: int | None = Field(default=None, ge=0)
    personal_score: float | None = Field(default=None, ge=0, le=10)
    review: str | None = None
    watched_at: str | None = None
    favorite: bool | None = None
    is_public: bool | None = None
    sort: int | None = None


class AcgImportRequest(AcgPersonalFields):
    pass


class AcgItemOut(AcgItemCreate):
    id: int
    created_at: datetime
    updated_at: datetime | None = None


class BangumiSearchItem(BaseModel):
    bangumi_id: int
    name: str
    name_cn: str = ""
    cover_url: str = ""
    summary: str = ""
    air_date: str = ""
    year: int = 0
    total_episodes: int = 0
    score: float = 0
    rank: int = 0
    tags: list[str] = Field(default_factory=list)
    source_url: str = ""
